File: leetcode/programmercarl/array/cpp/code/58.cpp
/*
 * @Description: 卡码网58题 区间和
 * @Author: 天涯过客
 * @Date: 2025-01-20 23:34:37
 */


#include<bits/stdc++.h>

using namespace std;

int main() {
    int n;
    cin>>n;
    vector<int> nums(n);
    vector<int> p(n+1);
    p[0] = 0;
    for(int i = 0;i<n;i++) {
        cin>>nums[i];
        p[i+1] = p[i] + nums[i];
    }
    
    
    int a,b;
    
    while(cin>>a>>b) {
        cout<<p[b+1]-p[a]<<endl;
    }
    return 0;
}

File: leetcode/programmercarl/array/java/_58.java
/*
 * @Description: 
 * @Author: 天涯过客
 * @Date: 2025-01-20 23:46:49
 */

import java.util.Scanner;


public class _58 {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        int n = scan.nextInt();
        int[] nums = new int[n];
        int[] p = new int[n+1];
        p[0] = 0;
        
        for(int i = 0;i<n;i++) {
            nums[i] = scan.nextInt();
            p[i+1] = p[i] + nums[i];
        }
        
        
        while(scan.hasNext()) {
            int a = scan.nextInt();
            int b = scan.nextInt();
            
            System.out.println(p[b+1] - p[a]);
        }
        scan.close();
    }
}
